Update release notes for oneDPL 2021.6.1 (#492)
oneapi-src/oneDPL: documentation/release_notes.rst

and provides high-productivity APIs aimed to minimize programming efforts of C++ developers
creating efficient heterogeneous applications.

New in 2021.6.1
===============

Fixed Issues
------------
- Fixed compilation errors with C++20.
- Fixed ``CL_OUT_OF_RESOURCES`` issue for Radix sort algorithm executed on CPU devices.
- Fixed crashes in exclusive_scan_by_segment, inclusive_scan_by_segment, reduce_by_segment algorithms applied to

>   device-allocated unified shared memory (USM).

Reviewer: I think wording "device-allocated unified shared memory (USM)" is confusing. I would suggest just "device-allocated memory (USM)".

Author: How about "applied to device-allocated USM" as was originally proposed by Alexey?

Reviewer: I agree.

Author: Done

and provides high-productivity APIs aimed to minimize programming efforts of C++ developers
creating efficient heterogeneous applications.

New in 2021.6.1
===============

Fixed Issues
------------

> - Fixed compilation errors happening when using C++20.

Reviewer 2: This looks more formal: 
```suggestion
- Fixed compilation errors occurring with the use of C++20.
```

> - Fixed compilation errors happening when using C++20.

Author: Changed to "Fixed compilation errors with C++20". @dmitriy-sobolev, are you ok with this change?

> - Fixed ``CL_OUT_OF_RESOURCES`` issue which used to occur during execution of Radix sort algorithm on CPU devices.

Reviewer 3: May be "Fixed ``CL_OUT_OF_RESOURCES`` issue for Radix sort algorithm executed on DPC++ CPU devices"?

Author: Applied Dmitriy's suggestion

> - Fixed an issue with the following algorithms used with DPC++ execution policies and the data located in the device memory (USM): ``exclusive_scan_by_segment``, ``inclusive_scan_by_segment``, ``reduce_by_segment``.

Reviewer 3: 1) the line is too long, needs to be split
2) what sort of an issue, how would it be observed?

Reviewer 3: Given the initial wording of "Fixed error in ``exclusive_scan_by_segment`` algorithm when used with unified shared device memory (USM) and number of items equals to 1", claiming this to be "added support" sounds too ambitious :)

How about "Fixed crashes in ``exclusive_scan_by_segment``, ``inclusive_scan_by_segment``, ``reduce_by_segment`` algorithms applied to device-allocated USM"?

and provides high-productivity APIs aimed to minimize programming efforts of C++ developers
creating efficient heterogeneous applications.

New in 2021.6.1
===============

New Features
------------

> - Added support of unified shared device memory (USM) for ``reduce_by_segment`` algorithm.

Author: Done

Fixed Issues
------------
- Fixed compilation errors happening when using C++20.

> - Fixed ``CL_OUT_OF_RESOURCES`` issue which used to occur during execution of Radix sort algorithm on CPU devices.

Author: I think, the article  is redundant here as this algorithm has a name.

> - Fixed error in ``exclusive_scan_by_segment`` algorithm when used with unified shared device memory (USM) and number of items equals to 1.

Author: Done

- Fixed error in ``inclusive_scan_by_segment`` algorithm when used with unified shared device memory (USM).
- Fixed error in ``reduce_by_segment`` algorithm when used with a DPC++ execution policy and unified shared device memory (USM).

Changes to Existing Features
----------------------------

> - Improved performance of ``shift_left``, ``shift_right`` algorithms on vector with ``seq``, ``par``, ``par_unseq``, ``unseq``

Reviewer: the wording should be re-worked... 
But, essentially the change about usage contiguous iterators is not completed in general. It fixes just  less 1% of parallel algorithms.. So, I would removed this input at all. Other words, I would removed lines 26-29

Reviewer 3: If we measured the performance improvement and know it's substantial, it might be worth mentioning even for a couple of algorithms. If we just assume improved performance and do not know how big the improvement is, I agree with Mikhail that it makes sense to first do general improvements for contiguous iterators (because the user value of this small improvement seems negligible).

Reviewer 2: Let's remove the note. 
```suggestion
```
#376 affects shift algorithms used with contiguous iterators and ``seq`` policy only. The performance improvement has not been measured and the user value seems too small for the note to be mentioned.

Author: Removed.
